es/es-server/src/main/java/org/elasticsearch/index/engine/Engine.java

```diff
@@ -1130,6 +1137,11 @@ public Index(Term uid, ParsedDocument doc, long seqNo, long primaryTerm, long ve
             this.ifPrimaryTermMatch = ifPrimaryTermMatch;
         }
 
+        Index(Term uid, long primaryTerm, ParsedDocument doc, long version) {
```

Comment:
same here, only used in test. I'd prefer not to add test-only code in general if not needed